### PR TITLE
Enhance election and survey reveal settings and reveal stage, and disable experiment modification by non-creators.

### LINKED ISSUE #104: When creating a "leadership election" module, experimenters should have an option to toggle on/off "vote for oneself"
And this should be toggled to "on" for the Lost at Sea experiment.

### PR DESCRIPTION
**Pull request by cjqian**
Fixes #104 #274 #273 #230

image

### CODE REVIEW

**Comment by cjqian** [comment]
Survey rendering:
image

**Review by vivtsai — COMMENTED** [review]
Hi! Let's merge the election/reveal changes, but roll back the "disable experiment modification" commit and fix #266 in a follow-up. I think we want to allow readers to view the experiment config, just not save changes or delete that experiment. Happy to discuss offline too.

File: utils/src/stages/stage.ts [hunk 1]
```diff
@@ -79,7 +87,8 @@ export interface StageProgressConfig {
 
 export type StageConfig =
   | ChatStageConfig
-  | ElectionStageConfig
+  | ItemRankingStage
```
Comment (vivtsai) [inline utils/src/stages/stage.ts:90]: FYI: Use `RankingStageConfig`, not `ItemRankingStage | ParticipantRankingStage`. Fixed in new PR.

**Comment by vivtsai** [comment]
Updated version: https://github.com/PAIR-code/deliberate-lab/pull/279

**Pull request closed without merging**